packages/assets-controllers/src/MultichainBalancesController/MultichainBalancesController.ts

```diff
@@ -186,8 +186,11 @@ export class MultichainBalancesController extends BaseController<
       const account = this.#getAccount(accountId);
 
       if (account.metadata.snap) {
-        const scope = getScopeForAccount(account);
-        const assetTypes = NETWORK_ASSETS_MAP[scope];
+        const assetsState = this.messagingSystem.call(
+          'MultichainAssetsController:getState',
+        );
+
+        const assetTypes = assetsState.accountsAssets[accountId] ?? [];
 
```

Comment:
Nice :D Should we also subscribe to assets state changes? to update the balances
```
    this.messagingSystem.subscribe(
      'MultichainAssetsController:stateChange',
      async (newAssets) => {
        //update balance for new assets
      },
    );
```

Reply:
I wonder if we should expose a `getAssetsForAccount` action on the `MultichainAssetsController` too? Feels like calling `:getState` is a bit too much here? 😅

(But we can probably start with that for now)

Reply:
and since we already have a mapping there for: account ID <> assets, that should be pretty straightforward too